frontend/dockerui/namedcontext.go

```diff
@@ -208,46 +214,64 @@ func (bc *Client) namedContextRecursive(ctx context.Context, name string, nameWi
 			if len(dt) != 0 {
 				excludes, err = dockerignore.ReadAll(bytes.NewBuffer(dt))
 				if err != nil {
-					return nil, nil, errors.Wrapf(err, "failed parsing %s", DefaultDockerignoreName)
+					return nil, errors.Wrapf(err, "failed parsing %s", DefaultDockerignoreName)
 				}
 			}
 		}
-		st = llb.Local(vv[1],
-			llb.WithCustomName("[context "+nameWithPlatform+"] load from client"),
-			llb.SessionID(bc.bopts.SessionID),
-			llb.SharedKeyHint("context:"+nameWithPlatform),
-			llb.ExcludePatterns(excludes),
-		)
-		return &st, nil, nil
+
+		// This is a weird invocation and requires an explanation.
+		// This slice is referenced and modified outside of this function, but it is
+		// read from the anonymous function passed to async. In order to ensure we're
+		// referencing the same slice, we allocate a pointer to a slice rather than just
+		// the slice which would be passed by value.
+		customOpts := new([]llb.LocalOption)
```

Comment:
I don't think this needs to be a pointer. You can just do:
```
var nc NamedContext
st = llb.Async(func() { if nc.localOpts {} })
nc.State = st
return &nc
```

Reply:
Originally this was going to be a slice that was modified. I think that doesn't really make sense anymore so I'll switch this up. It may be good to make it a bit more explicit though that the pointer is the one being captured. I think go should figure it out automatically, but just in case it would probably be better to do `nc := &NamedContext{}`.